src/three-components/TextureUtils.ts

```diff
@@ -406,28 +409,20 @@ void main() {
    * returns a texture of the prefiltered mipmapped radiance environment map
    * to be used as environment maps in models.
    */
-  pmremPass(texture: Texture, samples?: number, size?: number):
-      WebGLRenderTarget {
-    const generator = new PMREMGenerator(texture, samples, size);
-    generator.update(this.renderer);
-
-    const packer = new PMREMCubeUVPacker(generator.cubeLods);
-    packer.update(this.renderer);
-
-    const renderTarget = packer.CubeUVRenderTarget;
+  pmremPass(target: WebGLRenderTargetCube): WebGLRenderTarget {
+    const cubeUVTarget = this[$PMREMGenerator].update(target);
 
-    generator.dispose();
-    packer.dispose();
-
-    (renderTarget.texture as any).userData = {
+    (cubeUVTarget.texture as any).userData = {
       ...userData,
       ...({
-        url: (texture as any).userData ? (texture as any).userData.url : null,
+        url: (target.texture as any).userData ?
+            (target.texture as any).userData.url :
+            null,
         mapping: 'PMREM',
       })
     };
 
-    return renderTarget;
+    return cubeUVTarget;
```

Comment:
If I understand the implementation correctly, this means you are sharing the same render target instance for _all_ environment maps. That probably explains why you were only getting one working environment map.